tests/testthat/test_function-filtering.R

```diff
@@ -0,0 +1,112 @@
+xmse_full <- loadXcmsData("xmse")
+
+test_that("RsdFilter", {
+    # Test default parameters
+    filter <- RsdFilter()
+    expect_is(filter, "RsdFilter")
+    expect_equal(filter@threshold, 0.3)
+    expect_equal(filter@qcIndex, integer())
+    expect_equal(filter@na.rm, TRUE)
+    expect_equal(filter@mad, FALSE)
+
+    #test with XcmsExperiment object
+    filter <- RsdFilter(qcIndex = sampleData(xmse_full)$sample_type == "QC")
+    filtered_xmse <- filterFeatures(xmse_full, filter)
+    expect_lte(nrow(featureDefinitions(filtered_xmse)),
+               nrow(featureDefinitions(xmse_full)))
+
+    #test error message when index too small of too large
+    filter_1 <- RsdFilter(qcIndex = integer())
+    filter_2 <- RsdFilter(qcIndex = c(sampleData(xmse_full)$sample_type == "QC", TRUE))
+    expect_error(filterFeatures(xmse_full, filter_1))
+    expect_error(filterFeatures(xmse_full, filter_2))
+
+    #test in SummarizedExperiment object
+    res <- quantify(xmse_full)
+    filter <- RsdFilter(qcIndex = res$sample_type == "QC")
+    filtered_res <- filterFeatures(res, filter)
+    expect_lte(nrow(filtered_res), nrow(res))
+
+    #test same amount of features in sumexp and xcmsobjectg
+    expect_equal(nrow(featureDefinitions(filtered_xmse)),
+               nrow(filtered_res)) # this is not equal, normal ??
```

Comment:
hm, no, this has to be the same, if the same filter is applied.

Reply:
but it's working now, no? otherwise the unit tests would fail. so maybe remove the comment?